.github/workflows/pytest.yml

```diff
@@ -0,0 +1,192 @@
+name: Reusable WF - Run Pytest
+
+on:
+  workflow_call:
+    inputs:
+      platform:
+        required: true
+        type: string
+      platform-pretty:
+        required: true
+        type: string
+      config-fp:
+        required: false
+        type: string
+      config-pretty:
+        required: false
+        type: string
+        default: "[no config]"
+      python-version:
+        required: true
+        type: string
+      upstream-details:
+        required: false
+        type: string
+      workers:
+        required: false
+        type: string
+        default: "16"
+      pmlb-cache:
+        required: false
+        type: string
+        default: "false"
+      telemetry:
+        required: false
+        type: string
+        default: "true"
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+
+  test:
+    name: Pytest (${{ inputs.python-version }}, ${{inputs.platform-pretty}}, ${{ inputs.config-pretty }})
+    runs-on: ${{ inputs.platform }}
+    steps:
+
+      - uses: actions/checkout@v4
+
+      - name: Collect Workflow Telemetry
+        if: inputs.telemetry == 'true'
+        uses: catchpoint/workflow-telemetry-action@v1
+        with:
+          comment_on_pr: off
+          proc_trace_chart_show: off
+          proc_trace_table_show: off
+
+      - name: Pull GitHub cache
```

Comment:
How does this cache work? Where does it originally get set?

Reply:
This is really just for BDD since it's the only project that needs the PMLB cache. It's set in its `build.yml`, but didn't feel like I should make it a part of the template since it will likely never be used for any other project.

In BDD, though, if the cache doesn't exist than it goes through the process with Docker to download it, and then uploads to the cache (on a Linux runner). And then Windows/MacOS can access the cache here.

Reply:
This is set in the `build` job for BDD.